src/infill.cpp

```diff
@@ -82,59 +82,55 @@ void Infill::generate(std::vector<VariableWidthLines>& toolpaths,
-        // (Can't use getContours here, because only _some_ of the lines Arachne has generated are needed.)
-        Polygons gap_filled_areas;
-        for (const auto& var_width_line : gap_fill_paths)
+    // It does not make sense to print a pattern in a small region. So the infill region
+    // is split into a small region that will be filled with walls and the normal region
+    // that will be filled with the pattern. This split of regions is not needed if the
+    // infill pattern is concentric or if the small_area_width is zero.
+    if (pattern != EFillMethod::CONCENTRIC && small_area_width > 0)
+    {
+        // Split the infill region in a narrow region and the normal region.
+        Polygons small_infill = inner_contour;
+        inner_contour = inner_contour.offset(-small_area_width / 2).offset(small_area_width / 2);
+        small_infill = small_infill.intersection(small_infill.difference(inner_contour));
+        small_infill = Simplify(max_resolution, max_deviation, 0).polygon(small_infill);
+
+        // Small corners of a bigger area should not be considered narrow and are therefore added to the bigger area again.
+        for (PolygonsPart& small_infill_part : small_infill.splitIntoParts())
         {
-            VariableWidthLines thin_walls_only;
-            for (const auto& extrusion : var_width_line)
+            if (small_infill_part.offset(-infill_line_width / 2).offset(infill_line_width / 2).area() < infill_line_width * infill_line_width * 10)
             {
-                if (extrusion.is_odd && extrusion.inset_idx == 0)
+                if (! inner_contour.intersection(small_infill_part.offset(infill_line_width / 4)).empty())
                 {
-                    Polygon path;
-                    for (const auto& junction : extrusion.junctions)
-                    {
-                        path.add(junction.p);
-                    }
-                    if (path.polygonLength() >= infill_line_width * 4) // Don't fill gaps that are very small (with paths less than 2 line widths long, 4 back and forth).
-                    {
-                        gap_filled_areas.add(path);
-                        if (fill_gaps)
-                        {
-                            thin_walls_only.push_back(extrusion);
-                        }
-                    }
+                    small_infill = small_infill.difference(small_infill_part);
+                    inner_contour = inner_contour.unionPolygons(small_infill_part);
                 }
             }
-            if (! thin_walls_only.empty())
+        }
+
+        // Fill narrow area with walls.
+        const size_t narrow_wall_count = std::round(small_area_width / infill_line_width) + 1;
+        WallToolPaths wall_toolpaths(small_infill, infill_line_width, narrow_wall_count, 0, settings);
+        std::vector<VariableWidthLines> small_infill_paths = wall_toolpaths.getToolPaths();
+        if (! small_infill_paths.empty())
```

Comment:
You can loose this if-statement, if it's empty then the for loop will not be executed anyways.